pkg/apis/pipeline/v1alpha1/pipeline_types.go

```diff
@@ -43,6 +43,10 @@ type PipelineSpec struct {
 	// Results are values that this pipeline can output once run
 	// +optional
 	Results []PipelineResult `json:"results,omitempty"`
+	// Finally declares the list of Tasks that execute just before leaving the Pipeline
+	// i.e. either after all Tasks are finished executing successfully
+	// or after a failure which would result in ending the Pipeline
+	Finally []PipelineTask `json:"finally,omitempty"`
```

Comment:
We shouldn't add anything on `v1alpha1` now

Reply:
hey @vdemeester I have deleted `finally` from `v1alpha1`, have also added a unit test validating `convertFrom`, it should produce `ConversionError` while converting `v1beta1` `Pipeline` with `finally` to `v1alpha1`, let me know if the whole conversion related changes looks fine